### PR TITLE
New EVENT megafauna.

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

Adds a new megafauna mob, mainly for admin events.
Its main ability is to revive dead humans to join its side.
When it revives 12th human, it gains an ability to start the "rapture", which gives every previously converted(except for 12th) person a set of strong armor, unique weapon and a recall spell, as well as teleporting them to the megafauna mob.

apostle

Work In Progress.

## Why It's Good For The Game

Because I felt like there is not enough buttons to press to annoy people.

To me, it seems like a good event/gamemode(in future, if I ever manage to make it balanced).

## Changelog
:cl:
add: Heavens' wrath
/:cl:

### CODE REVIEW

File: code/FulpstationCode/fulp_mobs/apostle.dm [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	attack_sound = 'sound/magic/mm_hit.ogg'
+	icon = 'icons/Fulpicons/64x64.dmi'
+	icon_state = "angel"
+	icon_living = "angel"
+	health_doll_icon = "angel"
+	faction = list("apostle")
+	friendly_verb_continuous = "stares down"
+	friendly_verb_simple = "stare down"
+	speak_emote = list("proclaims")
+	armour_penetration = 100
```
Comment (JohnFulpWillard) [inline code/FulpstationCode/fulp_mobs/apostle.dm:17]: Doesnt this seem too unfair for people who bothered to get armor? I believe the only other thing that penetrates armor fully is the staff of storms

File: code/FulpstationCode/fulp_mobs/apostle.dm [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+					H.regenerate_limbs()
+					H.regenerate_organs()
+					H.dna.species.GiveSpeciesFlight(H)
+					H.revive(full_heal = TRUE, admin_revive = FALSE)
+					H.grab_ghost(force = TRUE)
+					to_chat(H, "<span class='notice'>You feel protected by holy light!</span>")
+					// Giving the fancy stuff to new apostle
+					//H.mind.add_antag_datum(/datum/antagonist/apostle)
+					H.faction |= "apostle"
+					ADD_TRAIT(H, TRAIT_BOMBIMMUNE, SPECIES_TRAIT)
```
Comment (JohnFulpWillard) [inline code/FulpstationCode/fulp_mobs/apostle.dm:105]: So here they can give someone wings, then makes them immune to bombs, fire, lowpressure and cold, while also changing their faction?

**Review by JohnFulpWillard — COMMENTED** [review]
This looks really good in general, my suggestions would be to give it crusher loot in case any miner decides to fight it (could even just be the same thing you get from colossus), and just lowering its armor penetration. A variant of this without the healing and just the fighting could make it a good thing to fight as an actual megafauna, since we have a lack of tegu-only icebox fauna

File: code/FulpstationCode/fulp_mobs/apostle.dm (hunk 2 above)
Comment (EgorDinamit) [reply code/FulpstationCode/fulp_mobs/apostle.dm:105]: It is supposed to switch them to the side of sentient mob. So far it wouldn't work as normal megafauna.

Comment (JohnFulpWillard) [reply code/FulpstationCode/fulp_mobs/apostle.dm:105]: So if theyre lucky enough to have this happen on a Captain/HoS/Geared up miner, they can just 'bone the entire station?

Comment (EgorDinamit) [reply code/FulpstationCode/fulp_mobs/apostle.dm:105]: It's up to whoever is playing as the "megafauna". However, they are being given an absolutely new set of items when the number of apostles approaches 12.

File: code/FulpstationCode/fulp_mobs/apostle.dm [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	loot = list(/obj/item/gun/magic/staff/chaos) // To be replaced later.
+	gps_name = "Pure Signal"
+	deathmessage = "evaporates in a moment, leaving heavenly light and feathers behind."
+	deathsound = 'sound/tegu_sounds/mob/apostle_death.ogg'
+	attack_action_types = list(/datum/action/innate/megafauna_attack/holy_revival,
+							   /datum/action/innate/megafauna_attack/fire_field)
+	small_sprite_type = /datum/action/small_sprite/megafauna/tegu/angel
+	var/holy_revival_cooldown = 20 SECONDS
+	var/holy_revival_cooldown_base = 20 SECONDS
+	var/holy_revival_damage = 10 // Amount of damage OR heal, depending on target.
```
Comment (Samuuu3) [inline code/FulpstationCode/fulp_mobs/apostle.dm:38]: might want to increase this or make them attack faster

**Review by Samuuu3 — APPROVED** [review]
I really like this. Only other thing I would change would maybe making them move a bit faster and having more damage since people can just run.

File: code/FulpstationCode/fulp_mobs/apostle.dm (hunk 3 above)
Comment (EgorDinamit) [reply code/FulpstationCode/fulp_mobs/apostle.dm:38]: It's being increased with each conversion, but I suppose it can be a bit faster.

Comment (Samuuu3) [reply code/FulpstationCode/fulp_mobs/apostle.dm]: First kill with no buddies is probably the hardest

Comment (EgorDinamit) [reply code/FulpstationCode/fulp_mobs/apostle.dm]: Changed in update-v3

File: code/FulpstationCode/fulp_mobs/apostle.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/FulpstationCode/fulp_mobs/apostle.dm]: Changed.

File: ModularTegustation/tegu_items/apostle_items.dm [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+/obj/item/clothing/suit/armor/apostle
+	name = "paradise lost"
+	desc = "Your armor, to protect the holy one."
+	worn_icon = 'ModularTegustation/Teguicons/teguclothing_worn.dmi'
+	icon_state = "apostlearmor"
+	item_flags = DROPDEL
+	heat_protection = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
+	body_parts_covered = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
+	armor = list(MELEE = 80, BULLET = 60, LASER = 60, ENERGY = 80, BOMB = 100, BIO = 100, RAD = 90, FIRE = 100, ACID = 95, WOUND = 25)
```
Comment (JohnFulpWillard) [inline ModularTegustation/tegu_items/apostle_items.dm:46]: Does this not seem too strong? Pretty sure this is ERT-level.

Comment (EgorDinamit) [reply ModularTegustation/tegu_items/apostle_items.dm:46]: It's basically the same thing as summoning Nar'Sie, albeit with a small chance of winning.

**Comment by EgorDinamit** [comment]
Fuck it. Less go.

**Pull request merged**